SpatialGDK/Documentation/content/non-unreal-server-worker-types.md

```diff
@@ -1,75 +1,85 @@
+>**Note:** Your external SpatialOS components must have an ID between 1000 and 2000 to be registered by the pipeline.
 
-#### Send data
-You set up your game to send SpatialOS component updates, command requests, and command responses directly using the `SpatialWorkerConnection.h` public methods:
+#### Using the code generator
 
-* `void SendComponentUpdate(Worker_EntityId EntityId, const Worker_ComponentUpdate* ComponentUpdate);`
-* `Worker_RequestId SendCommandRequest(Worker_EntityId EntityId, const Worker_CommandRequest* Request, uint32_t CommandId);`
-* `void SendCommandResponse(Worker_RequestId RequestId, const Worker_CommandResponse* Response);`
+The external schema code generator is a tool in the GDK that generates C++ code in your game module given a specific directory containing schema files. The generated code includes classes representing each external schema component, enumeration and type, and an `ExternalSchemaInterface` class for sending updates and registering callbacks for receiving network operations. The `ExternalSchemaInterface` generates methods for each component and command within a component in your external schema:
 
-You access these methods via a reference to the net driver, as shown below:
+* For each component, the `ExternalSchemaInterface` will contain methods for `SendComponentUpdate`, `OnAddComponent`, `OnComponentUpdate`, `OnAuthorityChange`, and `OnRemoveComponent`.
+* For each command, the `ExternalSchemaInterface` will contain methods for `SendCommandRequest`, `SendCommandResponse`, `OnCommandRequest` and `OnCommandResponse`.
 
-`SpatialWorkerConnection connection = Cast<USpatialNetDriver>(World->GetNetDriver())->Connection;`
+##### Setting up
 
-There is a basic example in the _Examples_ section below. For more examples of how to construct component updates, command requests, and more, see the SpatialOS documentation on  [serialization in the Worker SDK in C’s API](https://docs.improbable.io/reference/latest/capi/serialization).
+Below is a summary of the steps required to use the external schema code generator. There is a more complete example in the [examples section below](#using-the-code-generator-example).
 
-#### Receive data
+* Run the `ExternalSchemaCodegen.bat` file as described in the [helper scripts documentation]({{urlRoot}}/content/helper-scripts) to generate code from a directory containing external schema files.
+* [Regenerate](https://wiki.unrealengine.com/Generate_Visual_Studio_Project) the Visual Studio project files.
+* Instantiate the `ExternalSchemaInterface` class inside your game module. If you want to ensure that the SpatialOS worker connection registers your callbacks to receive initial network operations, you need to register the callbacks inside your game instance's `OnConnected` event callback.
```

Comment:
As a user, I don't understand what "initial network operations" or why would I want to ensure to receive those.

Reply:
@alastairdglennie ☝️

Reply:
```suggestion
* Instantiate the `ExternalSchemaInterface` class inside your game module. If you want to ensure that the SpatialOS worker connection registers your callbacks to receive the network operations that are sent as soon as your worker connects, you need to register the callbacks inside your game instance's `OnConnected` event callback.
```

Reply:
@improbable-davidrodriguez ☝️